Jb59383 devel

lbt opened the pull request:
Include the Sequel adaptor and support connection to a postgress DB
Improve config handling
Update gems

lbt committed:
Add ruote-sequel as possible backend

Signed-off-by: David Greaves <david.greaves@jolla.com>

lbt committed:
Update bunny to 2.19 and deps

bunny -> sorted_set -> (rbtree and set)

Signed-off-by: David Greaves <david.greaves@jolla.com>

lbt committed:
Update concurrent-ruby to 1.1.10

Signed-off-by: David Greaves <david.greaves@jolla.com>

lbt committed:
Add sequel for ruote-sequel

Signed-off-by: David Greaves <david.greaves@jolla.com>

lbt committed:
Added sequel to the build

Signed-off-by: David Greaves <david.greaves@jolla.com>

lbt committed:
Support the Sequel storage and handle config in boss/config.rb

Signed-off-by: David Greaves <david.greaves@jolla.com>

lbt committed:
Use BOSS.read_config

Signed-off-by: David Greaves <david.greaves@jolla.com>

lbt committed:
Move AMQP connection to BOSS.connect_to_amqp

Signed-off-by: David Greaves <david.greaves@jolla.com>

lbt committed:
Use the BOSS.read_config in the tools

Signed-off-by: David Greaves <david.greaves@jolla.com>

lbt committed:
Missed some $dashboard values

Signed-off-by: David Greaves <david.greaves@jolla.com>

lbt committed:
Correct use of route-fs

Signed-off-by: David Greaves <david.greaves@jolla.com>

lbt committed:
Improve error information

Signed-off-by: David Greaves <david.greaves@jolla.com>

lbt committed:
config is loaded after all other boss elements

Signed-off-by: David Greaves <david.greaves@jolla.com>

lbt committed:
Correct missed Boss -> BOSS

Signed-off-by: David Greaves <david.greaves@jolla.com>

lbt committed:
The base get_msgs() api doesn't allow an arg

Signed-off-by: David Greaves <david.greaves@jolla.com>

lbt committed:
Packaging and ad-hoc changes

Signed-off-by: David Greaves <david.greaves@jolla.com>

keto commented on .gitmodules:
```diff
@@ -4,3 +4,9 @@
+[submodule "ruote-sequel"]
+	path = ruote-sequel
+	url = https://github.com/lbt/ruote-sequel
+[submodule "ruote-redis"]
```
The configuration does not support the redis storage, so I'm not sure if it makes sense to add the module yet.

keto commented on Makefile:
```diff
@@ -3,18 +3,21 @@ bundle_dir = $(DESTDIR)$(target)
 	cd ruote-kit; gem build ruote-kit.gemspec
 	mv ruote-kit/ruote-kit-*.gem vendor/build/gems/
 
-install: vendor/build
```
The point of having the install in the Makefile was to make local testing and development easier. I would keep it here unless it is causing some issues. I'm also not sure how moving it to spec makes things any simpler like the commit message suggests.

lbt replied:
I fixed up the issue in osc by fixing the Makefile to not depend on the vendor/build dir instead

keto commented on boss.conf:
```diff
@@ -1,6 +1,9 @@
+# To use the old fs_storage
 db_path = /var/spool/boss/boss_ruote_db
+# Use the new sequel store
+# if db is specified it takes priority
```
I think this might cause confusion. If both are defined it should probably throw an error.
Or there could be just one option db, which takes file:// or postgres:// url. Not sure if e really need to care about the backwards compatibility for old configs.

keto commented on boss.conf:
```diff
@@ -1,6 +1,9 @@
+db = postgres://bossdb:boss@db/boss
 debug = false
 viewer_port = 9292
-viewer_address = 127.0.0.1
```
I would keep the default as localhost only.

keto commented on rpm/boss.spec:
```diff
@@ -1,5 +1,5 @@
 Name: boss
-Version: 0.11.0
+Version: 0.11.0+eos.20221111091745.7.gccc62f0
```
oops?

lbt committed:
Update Makefile so it always does a build for all

Previously, changes to the boss source would not be rebuilt in osc if
the vendor/build dir existed even if the source changed

Signed-off-by: David Greaves <david.greaves@jolla.com>
